### PR TITLE
OSDOCS-2485: Added CCCMO entry to the RH Operators reference assembly

### PR DESCRIPTION
**Pull request by jeana-redhat**
For https://issues.redhat.com/browse/OSDOCS-2485

Preview: 
https://deploy-preview-36669--osdocs.netlify.app/openshift-enterprise/latest/operators/operator-reference#cluster-cloud-controller-manager-operator_red-hat-operators

### CODE REVIEW

File: modules/cluster-cloud-controller-manager-operator.adoc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+== Purpose
+
+The Cluster Cloud Controller Manager Operator manages and updates the cloud controller managers deployed on top of {product-title}. The Operator is based on the Kubebuilder framework and `controller-runtime` libraries. It is installed via the Cluster Version Operator (CVO).
+
+It contains the following components:
+
+* Operator
+* Bootstrap manifest renderer
+* Installer based on static pods
+* Configuration observer
```
Comment (elmiko) [inline modules/cluster-cloud-controller-manager-operator.adoc:18]: i'm not sure if we consider the config sync controller part of this, perhaps @lobziik could add a little more info

**Review by elmiko — COMMENTED** [review]
i think this is mostly correct, just not sure about the sync controller

File: modules/cluster-cloud-controller-manager-operator.adoc [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+[discrete]
+== Purpose
+
+The Cluster Cloud Controller Manager Operator manages and updates the cloud controller managers deployed on top of {product-title}. The Operator is based on the Kubebuilder framework and `controller-runtime` libraries. It is installed via the Cluster Version Operator (CVO).
+
+It contains the following components:
+
+* Operator
+* Bootstrap manifest renderer
+* Installer based on static pods
```
Comment (JoelSpeed) [inline modules/cluster-cloud-controller-manager-operator.adoc:17]: I think we can drop these, we don't actually do either of these anymore, plans changed a little

Comment (lobziik) [reply modules/cluster-cloud-controller-manager-operator.adoc:17]: +1 enhancement part for bit more context: https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/out-of-tree-provider-support.md#bootstrap-changes

File: modules/cluster-cloud-controller-manager-operator.adoc (hunk 1 above)
Comment (lobziik) [reply modules/cluster-cloud-controller-manager-operator.adoc:18]: technically, it is a separate controller.

Maybe change this line a bit for being more specific? like `Cloud configuration observer`?

Our dev doc for bit context: https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/master/docs/dev/cloud-config-sync.md

**Comment by jeana-redhat** [comment]
@miyadav and/or @sunzhaohua2 this is ready for verification, PTAL

File: operators/operator-reference.adoc [hunk 3]
```diff
@@ -31,6 +31,7 @@ include::modules/ingress-operator.adoc[leveloffset=+1]
 include::modules/kube-apiserver-operator.adoc[leveloffset=+1]
 include::modules/kube-controller-manager-operator.adoc[leveloffset=+1]
 include::modules/cluster-kube-scheduler-operator.adoc[leveloffset=+1]
+include::modules/cluster-cloud-controller-manager-operator.adoc[leveloffset=+1]
```
Comment (bobfuru) [inline operators/operator-reference.adoc:34]: Should this move up to line 19 to be alphabetical? Seems the list mostly follows that convention.

File: modules/cluster-cloud-controller-manager-operator.adoc [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+//
+// * operators/operator-reference.adoc
+
+[id="cluster-cloud-controller-manager-operator_{context}"]
+= Cluster Cloud Controller Manager Operator
+
+[discrete]
+== Purpose
+
+The Cluster Cloud Controller Manager Operator manages and updates the cloud controller managers deployed on top of {product-title}. The Operator is based on the Kubebuilder framework and `controller-runtime` libraries. It is installed via the Cluster Version Operator (CVO).
```
Comment (bobfuru) [inline modules/cluster-cloud-controller-manager-operator.adoc:11]: Wondering if a note that this is in TP would be helpful? I'm not actually sure whether other Operators on this page are all GA or not, so I don't know that a TP note is needed here or not.

**Review by bobfuru — COMMENTED** [review]
Couple of minor questions but otherwise this LGTM.

File: modules/cluster-cloud-controller-manager-operator.adoc (hunk 4 above)
Comment (jeana-redhat) [reply modules/cluster-cloud-controller-manager-operator.adoc:11]: It's GA for ASH, TP for AWS, public Azure, and OpenStack. Attempted to thread that needle via release notes. But to clarify here, maybe a "this is only supported for ASH" note makes sense.

File: operators/operator-reference.adoc (hunk 3 above)
Comment (jeana-redhat) [reply operators/operator-reference.adoc:34]: Oh yeah I see it's mostly alpha, but the random O in the middle of the Cs threw me off. @adellape is the intent here to be alphabetical in general?

**Comment by miyadav** [comment]
lgtm , Thanks @jeana-redhat , adding "this is only supported for ASH" would be good.

**Comment by jeana-redhat** [comment]
Updates today:
- Added [a note](https://github.com/openshift/openshift-docs/pull/36669/files#diff-4bc06971b185a60ee5bd9834194ab46134427fd18aeaf9ea08e332de1a36f368R11) about supported status in ASH/Azure/AWS/OpenStack (would appreciate a look from @miyadav and a dev for this)
- Put this new Operator entry in alphabetical order
- Rearranged the couple Operators that were out of order (@adellape PTAL)

**Pull request merged**